src/adapters/networkhandler/mockNetworkHandler.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable import/no-dynamic-require */
+/* eslint-disable global-require */
+const path = require('path');
+const { getIntegrations } = require("../../routes/utils");
+// const { SUPPORTED_VERSIONS } = require("../../routes/utils/constants");
+
+const mockResponses = {};
+
+// Dynamically import the network handlers for all
+// the supported destinations
+const destinations = getIntegrations(path.resolve(__dirname, `../mocks`));
+
+destinations.forEach((dest) => {
+  mockResponses[dest] = require(`../mocks/${dest}/response.json`);
+});
+
+
+const sleep = async (sleepInMs) => new Promise((resolve) => { setTimeout(resolve, sleepInMs) });
+
+function mockNetworkHandler (type, originalNetworkHandler) {
+  const destMockResponses = mockResponses[type];
+
+  return function networkHandler() {
+    this.proxy = async (request) => {
+      if(!destMockResponses?.length || destMockResponses?.length === 0) {
+        // No destMockResponses available or if the destMockResponses is empty
+        return originalNetworkHandler?.proxy(request);
+      }
+      const randomIndex = Math.floor(Math.random() * (destMockResponses.length - 1));
```

Comment:
we can keep this optional unit testing

Reply:
Sorry I didn't understand what you're trying to say here. Can you explain a bit more ?

Reply:
If we use this for unit testing, can we keep sleep optional?

Reply:
Got it 👍🏼
Sleep is optional
Randomisation is not optional

Reply:
Will try to make it optional as well